pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -336,10 +353,15 @@ public long getLatestIngestionTimestamp() {
       // Null value vector
       MutableNullValueVector nullValueVector = _nullHandlingEnabled ? new MutableNullValueVector() : null;
 
+      String sourceColumn = metricsAggregators.containsKey(column) ? metricsAggregators.get(column).getFirst() : column;
+      ValueAggregator valueAggregator =
+          metricsAggregators.containsKey(column) ? metricsAggregators.get(column).getSecond() : null;
```

Comment:
Reduce the map lookups
```suggestion
      Pair<String, ValueAggregator> columnAggregatorPair = metricsAggregators.get(column);
      String sourceColumn = columnAggregatorPair != null ? columnAggregatorPair.getLeft() : null;
      ValueAggregator valueAggregator = columnAggregatorPair != null ? columnAggregatorPair.getRight() : null;
```